### PR TITLE
Upgrade to Infinispan 12.1.x

### PR DESCRIPTION
**Pull request by ryanemerson**
Infinispan 12.1.0.Final is due on the 02/04/21, we should upgrade the operator to CR2 images to ensure there are no issues.

### CODE REVIEW

**Comment by dmvolod** [comment]
@ryanemerson can you try #902 with 12.1? Probably this should work. I didn't try as image was not released with image-artifacts 2.1.3

**Comment by ryanemerson** [comment]
@dmvolod 12.1.x and 12.0.x has the same code for handling the exit status of batch commands, so it won't change anything. Further changes to the CLI are required.

**Comment by dmvolod** [comment]
> 12.1.x and 12.0.x has the same code for handling the exit status of batch commands, so it won't change anything. Further changes to the CLI are required.

Oh, but I'm wonder why 12.0.1 was working fine

**Comment by ryanemerson** [comment]
> Oh, sure, but I'm wonder why 12.0.1 was working fine

It's because of this commit I introduced https://github.com/infinispan/infinispan/commit/6a97209f4808e1d5f886d481c679e12aa83d61a8#diff-99bb88401742848e032fd6f51709415fb6be169a72d2e5d7fc44289255160d3c. It fixed the issue with invalid connection strings, but broke the batch status code.

**Review by dmvolod — APPROVED** [review]

**Comment by dmvolod** [comment]
@ryanemerson is there any chance to build Infinispan server image daily and run operator tests against this build. Does it make sense for early problems validation?

**Comment by dmvolod** [comment]
@ryanemerson is there any chance to build Infinispan server image daily and run operator tests against this build. Does it make sense for early problems validation?

**Comment by ryanemerson** [comment]
> @ryanemerson is there any chance to build Infinispan server image daily and run operator tests against this build. Does it make sense for early problems validation?

Infinispan currently provides a daily SNAPSHOT build of the server, however the problem is this is only deployed when the CI is green ... which is incredibly rare unfortunately, due to several flakey tests. So unless we change the snapshot release to deploy regardless of the CI result, it's not really possible todo this. @pruivo do you know what's required for this change?

**Comment by pruivo** [comment]
For upstream, as you said, it just needs to change the Jenkinsfile to deploy master every commit (or better, create a new job to deploy daily) 
For downstream, Osni is currently working on it since we plan to have operator nightlies for QE.

File: test/e2e/utils/constants.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ var (
 	RunSaOperator        = strings.ToUpper(constants.GetEnvWithDefault("RUN_SA_OPERATOR", "false"))
 	OperatorUpgradeStage = strings.ToUpper(constants.GetEnvWithDefault("OPERATOR_UPGRADE_STAGE", OperatorUpgradeStageNone))
 	CleanupInfinispan    = strings.ToUpper(constants.GetEnvWithDefault("CLEANUP_INFINISPAN_ON_FINISH", "true"))
-	ExpectedImage        = constants.GetEnvWithDefault("EXPECTED_IMAGE", "quay.io/infinispan/server:12.0")
+	ExpectedImage        = constants.GetEnvWithDefault("EXPECTED_IMAGE", "quay.io/infinispan/server:12.1")
 	NativeImageName      = constants.GetEnvWithDefault("NATIVE_IMAGE", "quay.io/infinispan/server-native:12.0")
```
Comment (rigazilla) [inline test/e2e/utils/constants.go:41]: @ryanemerson, this also needs update?

Comment (ryanemerson) [reply test/e2e/utils/constants.go]: Good spot @rigazilla. I have updated `server-native` to 12.1.

**Comment by rigazilla** [comment]
merged, thanks @ryanemerson

**Pull request merged**